src/AudioOutputSPDIF.cpp

```diff
@@ -0,0 +1,271 @@
+/*
+  AudioOutputSPDIF
+  
+  S/PDIF output via I2S
+  
+  Needs transciever from CMOS level to either optical or coaxial interface
+  See: https://www.epanorama.net/documents/audio/spdif.html
+
+  Original idea and sources: 
+    Forum thread dicussing implementation
+      https://forum.pjrc.com/threads/28639-S-pdif
+    Teensy Audio Library 
+      https://github.com/PaulStoffregen/Audio/blob/master/output_spdif2.cpp
+   
+  Adapted for ESP8266Audio
+
+  Copyright (C) 2020 Ivan Kostoski
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <Arduino.h>
+#ifdef ESP32
+  #include "driver/i2s.h"
+#else
+  #include <i2s.h>
+#endif
+#include "AudioOutputSPDIF.h"
+
+// BMC (Biphase Mark Coded) values (bit order reversed, i.e. LSB first)
+static const uint16_t spdif_bmclookup[256] = { 
```

Comment:
For the ESP8266, RAM is always tight (esp. w/MP3 decode).  Could we make this PROGMEM (a no-op on -32), please?

Reply:
I believed that 'static const' will go into flash anyway, as with ESP32, but I do not know enough for ESP8266. Changed as suggested.

Reply:
Thanks.  On the 8266 `static const` still has to go to HEAP because it can't do 8- or 16-b reads from ROM.  The PROGMEM attribute and the pgm_read_xxx macros take care of it.